src/ansys/fluent/core/systemcoupling.py

```diff
@@ -0,0 +1,119 @@
+"""Module containing System Coupling-related functionality."""
+
+from dataclasses import dataclass
+import os
+from typing import List
+import xml.etree.ElementTree
+
+
+class SystemCoupling:
+    """Wrap a System Coupling object, adding methods to discover more about the
+    System Coupling related setup and to help solving a System Coupling analysis.
+
+    Methods
+    -------
+    __getattr__
+    get_variables
+    get_regions
+    get_analysis_type
+    syc_connect
+    syc_solve
+    """
+
+    @dataclass
+    class Variable:
+        name: str
+        display_name: str
+        tensor_type: str
+        is_extensive: bool
+        location: str
+        quantity_type: str
+
+    @dataclass
+    class Region(object):
+        name: str
+        display_name: str
+        topology: str
+        input_variables: List[str]
+        output_variables: List[str]
+
+    def __init__(self, solver):
+        self._solver = solver
+        self.participant_type = "FLUENT"
+
+    # def __getattr__(self, attr):
+    #    return getattr(self.solver, attr)
+
+    def get_variables(self):
+        return self.__get_syc_setup()["variables"]
+
+    def get_regions(self):
+        return self.__get_syc_setup()["regions"]
+
+    def get_analysis_type(self):
+        if self._solver.setup.general.get_state()["solver"]["time"] == "steady":
+            return "Steady"
+        else:
+            return "Transient"
+
+    def syc_connect(self, host: str, port: int, name: str):
+        connect_command = f'(%sysc-connect-parallel "{host}" {port} "{name}")'
+        self.scheme_eval.exec((connect_command,))
```

Comment:
We would need to get the `scheme_eval` calls ripped out of here... Basically want to get it calling the API.

Reply:
Ok `scheme_eval` is gone now.